src/advanced-options.tsx

```diff
@@ -148,14 +146,12 @@ const AdvancedOptions = (
   const idemTokenId = `${formPrefix}${idemTokenName}`;
   return (
     <Stack spacing={4}>
-      {props.jobsView === JobsView.JobDetail && (
-        <TextField
+      {props.readonly && props.jobsView === JobsView.JobDetail && (
```

Comment:
We were using `jobsView === JobsView.JobDetail` to mean "make this read-only if we're looking at the detail view." Now that the detail view sets `props.readonly`, can we make this the only option? (Thinking specifically about the work I'm doing to create an alternate, mostly read-only create form for #209)

Reply:
The problem is because `model` is a union type, we need to check which member of the union it actually is to access member-specific fields (otherwise only common fields are available). And it's impossible to assign a type conditionally to `model` based on if `readonly` is set as ts is stripped out on compile. We could use `!` to assert the type of `model` but then we would be relying on convention ("if you set readonly, you promise to only use model of the appropriate type") vs type checking.